sw/device/lib/dif/dif_pinmux.h

```diff
@@ -0,0 +1,502 @@
+ */
+typedef enum dif_pinmux_sleep_toggle_result {
+  /**
+   * Indicates that the operation succeeded.
+   */
+  kDifPinmuxSleepToggleOk = kDifPinmuxOk,
+  /**
+   * Indicates some unspecified failure.
+   */
+  kDifPinmuxSleepToggleError = kDifPinmuxError,
+  /**
+   * Indicates that some parameter passed into a function failed a
+   * precondition.
+   *
+   * When this value is returned, no hardware operations occurred.
+   */
+  kDifPinmuxSleepToggleBadArg = kDifPinmuxBadArg,
+
+  /**
+   * Indicates that the operation is not permitted (locked).
+   */
+  kDifPinmuxSleepToggleLocked,
+} dif_pinmux_sleep_toggle_result_t;
+
+/**
+ * Enables the deep sleep mode of a particular pin.
+ *
+ * Note:
+ * After the chip has come back from the deep sleep, the mode of a pin
+ * persists until is explicitly cleared by `dif_pinmux_sleep_clear_state`.
+ * Calling `dif_pinmux_sleep_enable` will configure a pin mode for
+ * the next deep sleep, however it will not affect the present mode.
+ *
+ * @param pinmux A Pin Multiplexer handle.
+ * @param type Pin type (MIO or DIO).
+ * @param mode Pin deep sleep mode.
+ * @param index Pin index.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_pinmux_sleep_toggle_result_t dif_pinmux_sleep_enable(
+    dif_pinmux_t *pinmux, dif_pinmux_padring_pin_kind_t type,
+    dif_pinmux_sleep_mode_t mode, dif_pinmux_index_t index);
+
+/**
+ * Disables the deep sleep mode of a particular pin.
+ *
+ * Note:
+ * After the chip has come back from the deep sleep, the mode of a pin
+ * persists until is explicitly cleared by `dif_pinmux_sleep_clear_state`.
+ * Calling `dif_pinmux_sleep_enable` will configure a pin mode for
+ * the next deep sleep, however it will not change the current mode.
+ *
+ * @param pinmux A Pin Multiplexer handle.
+ * @param type Pin type (MIO or DIO).
+ * @param index Pin index.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_pinmux_sleep_toggle_result_t dif_pinmux_sleep_disable(
+    dif_pinmux_t *pinmux, dif_pinmux_padring_pin_kind_t type,
+    dif_pinmux_index_t index);
+
+/**
+ * Returns the state of a particular pin.
+ *
+ * When the pin is in the deep sleep mode, the mode persists until is explicitly
+ * cleared via `dif_pinmux_sleep_clear_state`.
+ *
+ * @param pinmux A Pin Multiplexer handle.
+ * @param type Padring pin type (MIO or DIO).
+ * @param index Pin index.
+ * @param[out] in_sleep_mode Pin state, `true` when in deep sleep mode.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_pinmux_result_t dif_pinmux_sleep_get_state(
+    const dif_pinmux_t *pinmux, dif_pinmux_padring_pin_kind_t type,
+    dif_pinmux_index_t index, bool *in_sleep_mode);
+
+/**
+ * Clears the deep sleep mode for a particular pin.
+ *
+ * When deep sleep mode is enabled for a pin, and the device has entered deep
+ * sleep mode; upon wake-up, the deep sleep this pin can be only cleared
```

Comment:
I would reword slightly as: ...the deep sleep status of this pin can only be cleared by calling this function.

Reply:
Done.